File: frontend/src/status.js
export const STATUS_COLORS = {
  UNDERWAY: '#22c55e',
  ANCHORED: '#f59e0b',
  MOORED: '#a855f7',
  STOPPED: '#ef4444',
  STALE: '#94a3b8',
  UNKNOWN: '#64748b',
}

export function statusColor(status) {
  return STATUS_COLORS[status] ?? STATUS_COLORS.UNKNOWN
}

File: stream_processor/status.py
def derive_status(nav_status, sog_knots) -> str:
    sog = sog_knots or 0
    if nav_status == 1:
        return "ANCHORED"
    if nav_status == 5:
        return "MOORED"
    if nav_status == 0 and sog == 0:
        return "STOPPED"
    if nav_status in (0, 8) and sog > 0:
        return "UNDERWAY"
    return "UNKNOWN"